File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
       int i=0;
        int j=0;
        
        while(j<word.length() && word[j]!=ch)
        {
            j++;
        }
        if(j==word.length())
        {
            return word;
        }
        while(i<j)
        {
            swap(word[i], word[j]);
            i++;
            j--;
        }
        return word;
    }
};

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        int start=0;
        char[] chArr=word.toCharArray();
        
        for(int i=0; i<chArr.length; i++)
        {
            if(chArr[i]==ch)
            {
                while(start<i)
                {
                    char a=chArr[start];
                    chArr[start]=chArr[i];
                    chArr[i]=a;
                    start++;
                    i--;
                }
                break;
            }
        }
        return new String(chArr);
    }
}